ale/isd_generate.py

```diff
@@ -185,11 +196,81 @@ def file_to_isd(
     else:
         usgscsm_str = ale.loads(file, props=props, verbose=log_level>logging.INFO, only_isis_spice=only_isis_spice, only_naif_spice=only_naif_spice)
 
-    logger.info(f"Writing: {isd_file}")
     isd_file.write_text(usgscsm_str)
 
+    if compress:
+        logger.info(f"Writing: {os.path.splitext(isd_file)[0] + '.br'}")
+        compress_json(isd_file)
+    else:
+        logger.info(f"Writing: {isd_file}")  
+
     return
 
+def write_json_file(isd_string, json_file):
+    """
+    Writes out json file with inputted isd string.
+    
+    Parameters
+    ----------
+    isd_string : str
+        The ISD as a JSON formatted string
+
+    json_file : str
+        Output json file path
+    """
+    with open(json_file, 'w') as fp:
+        json.dump(isd_string, fp, cls = AleJsonEncoder)
+
+def compress_json(uncompressed_json_file):
+    """
+    Compresses inputted .json file.
+    
+    Parameters
+    ----------
+    uncompressed_json_file : str
+        .json file path
+
+    Returns
+    -------
+    str
+        Compressed .br file path
+    """
+    if not os.path.splitext(uncompressed_json_file)[1] == '.json':
+        raise ValueError("Inputted file {} is not a valid .json file extension".format(uncompressed_json_file.split(".")[1]))
+    with open(uncompressed_json_file, 'rb') as f:
+        data = f.read()
+    with open(uncompressed_json_file, 'wb') as f:
+        f.write(brotli.compress(data))
+
+    os.rename(uncompressed_json_file, os.path.splitext(uncompressed_json_file)[0] + '.br')
+
+    return os.path.splitext(uncompressed_json_file)[0] + '.br'
+
+
+def decompress_json(compressed_json_file):
+    """
+    Decompresses inputted .br file.
+    
+    Parameters
+    ----------
+    compressed_json_file : str
+        .br file path
+
+    Returns
+    -------
+    str
+        Decompressed .json file path
+    """
+    if not os.path.splitext(compressed_json_file)[1] == '.br':
+        raise ValueError("Inputted file {} is not a valid .br file extension".format(compressed_json_file))
+    with open(compressed_json_file, 'rb') as f:
+        data = f.read()
+    with open(compressed_json_file, 'wb') as f:
+        f.write(brotli.decompress(data))
```

Comment:
I am confused by the decompression here. Is this read the compressed and then write the decompressed to disk? Would it make more sense to simply pipe the decompressed to stdout? That would be more bash-like.

Reply:
This is for situations where a user wants to revert to a readable JSON file saved locally. We could add a function to read the JSON from binary format, if you think that would be appropriate.